core/modules/simpletest/tests/upgrade/upgrade.test

```diff
@@ -207,7 +207,15 @@ abstract class UpgradePathTestCase extends BackdropWebTestCase {
     // Check if there still are pending updates.
     $this->backdropGet($update_url, array('external' => TRUE));
     $this->backdropPost(NULL, array(), t('Continue'));
-    if (!$this->assertText(t('No pending updates.'), 'No pending updates at the end of the update process.')) {
+
+    // Install new updates if any.
+    if ($this->assertText(t('Updates have been found that need to be applied'))) {
+      $this->backdropPost(NULL, array(), t('Apply pending updates'));
+      if (!$this->assertText(t('Updates were attempted'))) {
+        return FALSE;
+      }
+    }
+    elseif (!$this->assertText(t('No pending updates.'), 'No pending updates at the end of the update process.')) {
```

Comment:
I wondered if this required a separate issue in fact, but it was necessary for tests to pass. 

The current `upgrade.test` `performUpgrade()` runs on a database without `layout` module, so `layout_update_N()` doesnt run and the function expects that after a D7 upgrade there would be 'no pending upgrades'. So when layout installs with my new `layout_update_1000(),` `performUpgrade()` finds this and terminates the upgrade. This little code block anticipates this.

An alternative is to not call `performUpgrade(),` and to duplicate it in `layout.test` instead, but it seemed wiser to fix the `upgrade.test` version instead. 

What else could we do here?